client/gutenberg/editor/editor-document-head/index.jsx

```diff
@@ -0,0 +1,47 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import DocumentHead from 'components/data/document-head';
+import { getPostType } from 'state/post-types/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+
+export const EditorDocumentHead = ( { isNewPost, postType, postTypeObject, translate } ) => {
+	let title = '';
+	if ( 'post' === postType ) {
+		title = isNewPost ? translate( 'New Post' ) : translate( 'Edit Post' );
+	} else if ( 'page' === postType ) {
+		title = isNewPost ? translate( 'New Page' ) : translate( 'Edit Page' );
+	} else if ( !! postTypeObject ) {
+		title = isNewPost ? postTypeObject.labels.new_item : postTypeObject.labels.edit_item;
+	} else {
+		title = isNewPost ? translate( 'New' ) : translate( 'Edit' );
+	}
+
+	return <DocumentHead title={ title } />;
+};
+
+EditorDocumentHead.propTypes = {
+	isNewPost: PropTypes.bool,
+	postId: PropTypes.number,
+	postType: PropTypes.string,
+	postTypeObject: PropTypes.object,
+};
+
+const mapStateToProps = ( state, { postId, postType } ) => {
+	const siteId = getSelectedSiteId( state );
+	return {
+		isNewPost: ! postId,
+		postTypeObject: getPostType( state, siteId, postType ),
```

Comment:
This is the wrong post schema isn't it? We'll want to use the wp.data and the v2 posts, similar to: https://github.com/Automattic/wp-calypso/pull/27301/files#diff-16d357cdb3cf5543a9079d68a41f36d5R38